pkg/virt-controller/services/template_test.go

```diff
@@ -1460,9 +1460,23 @@ var _ = Describe("Template", func() {
 					Expect(err).ToNot(HaveOccurred())
 
 					if isLabelExpected {
-						Expect(pod.Spec.NodeSelector).To(HaveKeyWithValue("scheduling.node.kubevirt.io/tsc-frequency-123123", "true"))
```

Comment:
Can it be that we need a more complex logic here?

I think there needs to be at least an `or`:

```
scheduling.node.kubevirt.io/tsc-frequency-123123
OR
(
scheduling.node.kubevirt.io/tsc-frequency gt 123123-533000
AND
scheduling.node.kubevirt.io/tsc-frequency lt 123123+533000
)
```

I think this way, we can cover all nodes with inv-tsc which can scale to the right range as well, as the nodes which are in the range natively and cna't scale.

Reply:
Is there a way to do an `or` with node affinities? Doesn't the `tsc-frequency-xxx` label just mirror the value of the `tsc-frequency` one?

Reply:
> Doesn't the tsc-frequency-xxx label just mirror the value of the tsc-frequency one?

It works like this: 
 * Detect the lowest reported tsc frequency via checking `scheduling.node.kubevirt.io/tsc-frequency` of each node.
 * If the node has this frequency, OR can scale TSC, it gets labelled with `scheduling.node.kubevirt.io/tsc-frequency-123`
 * If new nodes join or leave the cluster, the minimum frequency may change. If the old frequency is not actively used by any VMI, the old label gets removed. Otherwise all nodes compatible with that frequency keep it until the last VM using it is gone.
 * The new minimum frequency is added.
 
 This means: VMIs stay migratable even when nodes are added or removed, since the old node labels stay if they are still in use.
 
 We have a few options here. I think the simplest one would be doing the following:
 
  1. changing the tsc-frequency-<number> label calculation so, that it does not choose the cluster-minimum, but considers in addition a range: minimum + 1066/2. This is then the frequency which we label on the node which are in that range.
  2. Optionally (can be added later without breaking anything): If a minimum freuquency is pre-selected, we may also have nodes with a frequency below the selected minimum, there we can actually look at the whole range: -1066 to selected to +1066 and choose then actually the frequency which includes the most amount  of compatible nodes.